File: longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        if(nums.size() == 0 || nums.size() == 1)
        {
            return nums.size();
        }


        set<int> s;
        vector<int> v;
        for(auto it : nums)
        {
            s.insert(it);
        }

        for(auto it : s)
        {
            v.push_back(it);
        }

        int N = v.size();
        int ans = 0;
        int tempLen = 1;
        int i = 1;

        while(i < N)
        {
            if(abs(v[i] - v[i - 1]) == 1)
            {
                tempLen++;
                i++;
            }
            else
            {
                i++;
                ans = max(ans , tempLen);
                tempLen = 1;
            }
        }

        //Since it wont consider after last comparison 
        ans = max(ans , tempLen);
        return ans;



        return s.size();
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        if(len(nums) == 0):
            return 0

        nums = list(set(nums))
        nums.sort()
        
        maxi = 1
        ans = 1
        prev = nums[0]
        for i in range(1 , len(nums)):
            if(nums[i] - prev == 1):
                prev = nums[i]
                maxi += 1
            
            else:
                prev = nums[i]
                ans = max(ans , maxi)
                maxi = 1
            

        return max(ans , maxi)